sdk/python/v1beta1/kubeflow/katib/utils/utils.py

```diff
@@ -129,3 +132,132 @@ class FakeResponse:
 
     def __init__(self, obj):
         self.data = json.dumps(obj)
+
+
+class SetEncoder(json.JSONEncoder):
+    def default(self, obj):
+        if isinstance(obj, set):
+            return list(obj)
+        if isinstance(obj, type):
+            return obj.__name__
+        return json.JSONEncoder.default(self, obj)
+
+
+def parameter_substitution(
+    parameters: Union[Dict[str, Any], "TrainingArguments", "LoraConfig"],  # noqa: F821
+    experiment_params: List[models.V1beta1ParameterSpec],
+    trial_params: List[models.V1beta1TrialParameterSpec],
+):
```

Comment:
@helenxie-bit Can you also add the return of this function ? 
```
-> Dict[str,str]

Reply:
Got it! I will add it.